openthread/Kconfig

```diff
@@ -24,4 +26,79 @@ config OPENTHREAD_LIBRARY_1_1
 
 endchoice
 
+# OpenThread configuration features sets from Nordic Semiconductor
+
+choice
```

Comment:
I don't quite get how does it work. We set some predefined values, which is ok and seemed to be missing as those are no longer configurable when building with library, but I can't find any connection with the libraries being selected by linker. For example I was expecting that OPENTHREAD_NORDIC_LIBRARY_MTD will be selected automatically when CONFIG_OPENTHREAD_MTD is selected (or the other way around i.e. select CONFIG_OPENTHREAD_MTD when OPENTHREAD_NORDIC_LIBRARY_MTD). Currently it is possible to select OPENTHREAD_FTD and OPENTHREAD_NORDIC_LIBRARY_MTD at the same time.

Reply:
Yes, that's possible, but those are different things:
- `OPENTHREAD_FTD/MTD` is an OpenThread stack configuration that defines the device's possible roles.
- `OPENTHREAD_NORDIC_LIBRARY_MASTER/FTD/MTD` are sets of predefined extra configurations for OpenThread intended to be certified as a pack in order to ease certification by inheritance for Nordic customers. The sets were already present in nRF5 SDK: https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.1.0%2Fthread_openthread_libraries.html

Actually if you look at the table it's not so strange to use `OPENTHREAD_MTD` with `OPENTHREAD_NORDIC_LIBRARY_FTD`, there could be some use case of an MTD needing to announce a Thread Service, for instance.

KRKNWK-6714 intends to update this in order to have a matching library for each of the `OPENTHREAD_NORDIC_LIBRARY_MASTER/FTD/MTD` options.